topics/ecology/tutorials/phylogeny-data-prep/tutorial.md

```diff
@@ -0,0 +1,332 @@
+{: .hands_on}
+
+# Genome annotation
+
+## Mask repetitive sequences
+Before we can annotate the genome, we will prepare the data by masking repetitive sequences in the genome.
+Repeat-rich regions can interfere with genome annotation tools. In this step we find and soft-mask repetitive regions in the genome. The annotation tool can then take this information into account ([see RepeatMasker tutorial for more details]({% link topics/genome-annotation/tutorials/repeatmasker/tutorial.md %})).
+We use **RepeatMasker**
+{% cite RepeatMasker %}, a program that screens DNA sequences for interspersed repeats and low complexity DNA sequences.
+This program has historically made use of [RepBase](https://www.girinst.org/repbase/update/index.html) ({%cite Kohany2006-ks %}), a service of the Genetic Information Research Institute, but this database in no longer open access. Instead, we will use [Dfam](https://www.dfam.org/home) ({%cite Storer2021 %}) an open collection of Transposable Element DNA sequence alignments,  HMMs derived from Repbase sequences and consensus sequences. For this reason, the annotation of repetitive sequences might be incomplete.
+
+RepeatMasker will only accept compact fasta headers. Before we can mask repetitive regions with RepeatMasker we must trim the NCBI long header (`BK006939.2 TPA_inf: Saccharomyces cerevisiae S288C chromosome V, complete sequence`) to leave only the accession number (`>BK006939.2`) by using a regular expression.
+> ### {% icon hands_on %} Hands-on: Mask repetitive sequences
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2) %} with the following parameters:
+>    - {% icon param-collection %} *"File to process"*: `output` (Input dataset collection)
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"Find pattern"*: `(>[^ ]+).+`
+>            - *"Replace with:"*: `\1`
+>
+>    > ### {% icon comment %} Comment
+>    >
+>    >`\1` replaces the text in the header with the first matched text in the header, the accession number in this case.
+>    {: .comment}
+>
+> 2. {% tool [RepeatMasker](toolshed.g2.bx.psu.edu/repos/bgruening/repeat_masker/repeatmasker_wrapper/4.1.2-p1+galaxy0) %} with the following parameters:
+>    - {% icon param-file %} *"Genomic DNA"*: `outfile` (output of **Replace Text** {% icon tool %})
+>    - *"Repeat library source"*: `DFam (curated only, bundled with RepeatMasker)`
+>        - *"Select species name from a list?"*: `No`
+>            - *"Repeat source species"*: `"Saccharomyces cerevisiae"`
+>
+>    > ### {% icon comment %} Comment
+>    >
+>    > In principal, all unique clade names occurring in [NCBI taxonomy database](https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html) can be used for species. Capitalization is ignored, multiple words need to bound by apostrophes. Not all "common" English names occur in the taxonomy database. Using Latin names is always safest.
```

Comment:
What do you mean here? Is it a comment related to RepeatMasker or how to prepare data for RepeatMasker?